dashboard/app/serializers/api/v1/pd/workshop_enrollment_serializer.rb

```diff
@@ -7,7 +7,9 @@ def user_id
   end
 
   def district_name
-    object.school_district.try(:name)
+    school_district_id = object.try(:school_info).try(:school_district_id)
+    school_district = SchoolDistrict.find(school_district_id) if school_district_id
+    school_district.name if school_district
```

Comment:
This seems like an awful lot of logic in the serializer. How about adding an association, or if that's too complex, a method with this logic, into the `Enrollment` model?